### PR TITLE
Connect uerdb

### PR DESCRIPTION
**Pull request by aaron-azarcon**
-> Connected user_table from database to the Sign up and Log in feature of the web app. 
-> There are placeholder redirect pages to indicate successful Sign Up and Log In.
->  Changes in the database have been made for this version of Sign Up and Log In. Specifically, table 'users' have changes in the 'user_addresses' column where it currently accepts NULL values since user addresses have yet to be implemented. The column 'user_contact_num' was modified to be an index and must have no duplicates. 
-> Folder Names were changed as to remove whitespaces to make file paths easier to work with.
-> error handling needs improvement

### CODE REVIEW

File: db/Albertos_db.sql [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+-- Table structure for table `users`
+--
+
+DROP TABLE IF EXISTS `users`;
+/*!40101 SET @saved_cs_client     = @@character_set_client */;
+/*!50503 SET character_set_client = utf8mb4 */;
+CREATE TABLE `users` (
+  `user_id` varchar(255) NOT NULL,
+  `user_fullname` varchar(255) NOT NULL,
+  `user_email` varchar(255) NOT NULL,
```
Comment (kedalisay) [inline db/Albertos_db.sql]: Remove the `user_email` field as there's no use for it within the website.

image

File: db_connections/database.php [hunk 2]
```diff
@@ -0,0 +1,19 @@
+<?php
+
+$host = "localhost";
+$dbname = "Albertos_database";
```
Comment (kedalisay) [inline db_connections/database.php]: The name of the database should be all lowercase (`albertos_database`) as reflected on the SQL file.

File: db_connections/database.php
Comment (kedalisay) [inline db_connections/database.php:1]: We can use `PDO` (PHP Data Objects) next time in place of `mysqli`. It's flexible as we can use almost any database with it. I'm planning to make most of our code conform to OOP so that you and I can understand what we're working on. 

We're going to abstract the functionalities into their own objects.

File: db/Albertos_db.sql [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+  `user_contact_num` varchar(255) NOT NULL,
+  `user_password` varchar(255) NOT NULL,
+  `user_addresses` longtext CHARACTER SET utf8mb4 COLLATE utf8mb4_bin NOT NULL CHECK (json_valid(`user_addresses`)),
+  PRIMARY KEY (`user_id`)
```
Comment (kedalisay) [inline db/Albertos_db.sql:190]: I believe you haven't updated the SQL file?

File: db_connections/process-signup.php [hunk 4]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+if( empty($_POST["fullname"])){
+    die("Full Name is required");
+}
```
Comment (kedalisay) [inline db_connections/process-signup.php:5]: When an error is encountered within the signup process, the website usually redirects the user back to the signup page and display the error there. That's the change we'll do soon.

We can convert the signup process to its own object.

File: db_connections/process-signup.php [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+$uniqueId = $fullname . uniqid();
+
+$sql = "INSERT INTO users (user_id, user_fullname, user_contact_num, user_password)
+        VALUES (?, ?, ?, ?)";
+
+$stmt = $mysqli->stmt_init();
+
+if( ! $stmt->prepare($sql)){
+    die("SQL Error: " . $mysqli->error);
+}
```
Comment (kedalisay) [inline db_connections/process-signup.php:23]: For internal server errors, we can display a generic 503 page for client users in production. 

Developers meanwhile, in both testing and production, can access and display the error in two ways:
1. Explicitly display it like in lines 21-23.
2. Log the error in a text file.

TL;DR Apache Server stores the error log file on the file path `/var/log/apache2` on Linux. For XAMPP, the dashboard has a button that can take you to the error log file within the XAMPP dashboard.

[For more answers on PHP error log files, Stack Overflow has a bunch.](https://stackoverflow.com/questions/5127838/where-does-php-store-the-error-log-php-5-apache-fastcgi-and-cpanel)

File: SignUp_Page/signup-success.html
Comment (kedalisay) [inline SignUp_Page/signup-success.html:1]: When signup is a success, the client user should be directed to the log-in page to input his/her credentials.

That's what we'll replace with this soon.

File: Login_Page/login-success.html
Comment (kedalisay) [inline Login_Page/login-success.html:1]: Upon successful log-in, the user should be directed back to the homepage. 

We'll replace this soon.

**Review by kedalisay — CHANGES_REQUESTED** [review]
TL;DR we'll follow the OOP paradigm when creating code for the website. You don't have to convert it right now as we'll plan it when we meet in-person or remotely again.

The rest of my comments you can look into it for change and info.
